### ISSUE TITLE
Bz2 library import error on macos, python3.7

### ISSUE THREAD

**Issue by narugo1992** [huggingface/huggingface_hub#1519]
### Describe the bug

When executing `from huggingface_hub import HfFileSystem, HfApi` in macOS with Python 3.7, the following error occurs:

```text
test/conftest.py:4: in <module>
    from huggingface_hub import HfFileSystem, HfApi
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/huggingface_hub/__init__.py:311: in __getattr__
    submod = importlib.import_module(submod_path)
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/huggingface_hub/hf_file_system.py:10: in <module>
    import fsspec
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/fsspec/__init__.py:12: in <module>
    from .compression import available_compressions
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/site-packages/fsspec/compression.py:2: in <module>
    from bz2 import BZ2File
../../../hostedtoolcache/Python/3.7.17/x64/lib/python3.7/bz2.py:19: in <module>
    from _bz2 import BZ2Compressor, BZ2Decompressor
E   ModuleNotFoundError: No module named '_bz2'
```

### Reproduction

See logs part.

The issue can be consistently reproduced on macOS runner of Github Actions.

### Logs

```shell
https://github.com/deepghs/tbsync/actions/runs/5314756187/jobs/9622324523#step:14:33

**This issue seems to only occur in Python 3.7 on macOS**.
```

### System info

```shell
The execution environment is GitHub Actions, and I cannot perform this operation. Additionally, I do not have an available macOS environment locally. Sorry for this.

`huggingface-hub`'s version is 0.15.1
```

**Comment by Wauplin** [comment]
Hi, thanks for reporting. I don't think this is related to `huggingface_hub` but more `fsspec` itself, isn't it? According [to their repo](https://github.com/fsspec/filesystem_spec/blob/master/setup.py#L56), it is supported only for Python>=3.8. To be honest I didn't know and thought it was >=3.7 as well. As Python 3.7 itself is coming to an end as well, I don't think we'll investigate time to check how to make it compatible. Maybe we could more gracefully handle the issue though.

would it be possible to run your github CI on macos Python>=3.8 instead?

**Comment by narugo1992** [comment]
@Wauplin 

This issue is indeed quite tricky. I have reviewed the relevant information on `fsspec`, but couldn't find any further details. Later, I tried testing it on some temporary macOS environments and found that the problem cannot be consistently reproduced on certain macOS systems. It may be related to the presence of `libbzip2` in the system during Python compilation and installation. Therefore, we will exclude the unit tests on macOS with Python 3.7 and inform users about this situation in the README or documentation.
